pkg/reporter/gh-annotations_reporter.go

```diff
@@ -12,27 +12,45 @@ type GHAnnotationsReporter struct {
 	hasPrintedError bool
 	stdout          io.Writer
 	stderr          io.Writer
+	level           VerbosityLevel
 }
 
-func NewGHAnnotationsReporter(stdout io.Writer, stderr io.Writer) *GHAnnotationsReporter {
+func NewGHAnnotationsReporter(stdout io.Writer, stderr io.Writer, level VerbosityLevel) *GHAnnotationsReporter {
 	return &GHAnnotationsReporter{
 		stdout:          stdout,
 		stderr:          stderr,
+		level:           level,
 		hasPrintedError: false,
 	}
 }
 
-func (r *GHAnnotationsReporter) PrintErrorf(msg string, a ...any) {
-	fmt.Fprintf(r.stderr, msg, a...)
-	r.hasPrintedError = true
+func (r *GHAnnotationsReporter) Errorf(format string, a ...any) {
+	if ErrorLevel <= r.level {
+		fmt.Fprintf(r.stderr, format, a...)
+		r.hasPrintedError = true
```

Comment:
This should always be set, regardless of if the error is actually printed anyway - it's used to determine if there were any errors that mean the exit code should not be 0.

Though I also wonder if we should be having a condition for this? it doesn't matter much here except maybe letting you simplify tests, but the only time I expect we might want to not print a fatal error would be if we supported a "silent" level, and even then tbh I'd be on the fence since we're talking about errors that completely exploded the tool.

Reply:
ah right this actually isn't a bug because the condition is always true - I do not like that though because it feels too easy to mistakenly bug out; i.e. if you change `ErrorLevel` to something else, then the only test that will fail is the `Reporter` one even though it actually breaks a bunch of other tests throughout the codebase which are using the default verbosity

Reply:
> This should always be set, regardless of if the error is actually printed anyway - it's used to determine if there were any errors that mean the exit code should not be 0.

Thanks for the review, and nice catch!

> Though I also wonder if we should be having a condition for this? 

Yeah I thought of this too but believed adding an if statement in these `Errorf()` implementations would avoid future problems if levels lower than `ErrorLevel` were introduced. Let me know if you feel differently!

Reply:
> Let me know if you feel differently!

I do very slightly for two reasons: 1. I really cannot imagine a situation where we'd not want to report an error even with a `--silent` flag I'd make a argument that the errors we're expecting to report with this shouldn't be silenced, and 2. GoLand flags this condition as always being true - those two combined ever-so-slightly push the needle for me towards removing the condition, because less IDE warnings is always nice (and it's technically slightly less code too, which I guess arguably is a third reason)

I won't block over this though, so up to you :)

Reply:
Makes sense to me, I'll make the necessary changes (though I'm not sure if the tests for `Errorf()` should also change?)

Reply:
Na the tests should be fine because they're testing the behaviour not the implementation - they shouldn't care about what logic actually goes on, just that the outcome is what's expected for their cases.